README.md
# AADC
Airport Arrival Demand Chart - VATSIM
# AADCv2
Massive update a ton of stuff check the changelog.


public/changelog.txt
---- AADC V2.1 ----
*AAR, moves with time.
*Increase reliability of main python program.
*Added an API for support with future IDS.
*Publishes current AAR for other applications to grab.
*Added temporary flight list in the console.
*Added the ChangeLog :)

---- AADC V2 ----
*Completely Reworked logic and aircraft calculations



main.py
#!/usr/bin/python3

import urllib.request
import json
import math
import datetime
import time

while(True):
    now = datetime.datetime.utcnow()
    hour = now.strftime("%H")
    min = int(now.strftime("%M"))
    minx = min / 0.6
    #print("Current time : " + now.strftime("%H%M") + "z")


    def distance(lat1, lon1):
        lat2, lon2 = 39.8616667, -104.6731667
        radius = 6371  # km

        dlat = math.radians(lat2 - lat1)
        dlon = math.radians(lon2 - lon1)
        a = math.sin(dlat / 2) * math.sin(dlat / 2) + math.cos(math.radians(lat1)) * math.cos(math.radians(lat2)) * math.sin(dlon / 2) * math.sin(dlon / 2)
        c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
        d = radius * c * 0.5399568035

        return d


    def ERTCalc(flight):
        # Knots are NM/Hour
        flight.update({"distance": distance(flight["latitude"], flight["longitude"])})
        ERT = flight["distance"] / (int(flight["planned_tascruise"]))
        HOURERT = str(round(ERT, 2)).split(".")[0]
        MINERT = round(int(str(round(ERT, 2)).split(".")[1])*.6)
        flight.update({"ESTHOURERT": HOURERT})
        flight.update({"ESTMINERT": MINERT})

    def GroundCalculation(flight):
        if int(flight["planned_tascruise"]) < 50:
            ##No TAS retarded Pilot
            flight["planned_tascruise"] = 350
        ERTCalc(flight)
        EstimatedATHour = 0
        EstimatedATMin = 0

        if flight["planned_deptime"] == 0:
            ##No Dep Time retarded pilot
            flight["planned_deptime"] = now.strftime("%H%M") + 30

        # Verify Planned Departure Time
        if int(flight["planned_deptime"]) < (int(now.strftime("%H%M")) - 30) or int(flight["planned_deptime"]) > (int(now.strftime("%H%M")) + 30):
            flight["planned_deptime"] = int(now.strftime("%H%M")) + 30
            if(flight["planned_deptime"]%100 > 60):
                flight["planned_deptime"] = flight["planned_deptime"] + 40
        if int(flight["planned_deptime"]) < 100:
            EstimatedATHour = flight["ESTHOURERT"]
            EstimatedATMin = flight["planned_deptime"] + flight["planned_deptime"]
        else:
            EstimatedATHour = int(flight["ESTHOURERT"]) + int(str(flight["planned_deptime"])[:-2])
            EstimatedATMin = int(flight["ESTMINERT"]) + int(str(flight["planned_deptime"])[-2:])
        while(int(EstimatedATMin) > 59):
            EstimatedATMin = int(EstimatedATMin) - 60
            EstimatedATHour = int(EstimatedATHour) + 1
        if(int(EstimatedATHour) > 23):
            EstimatedATHour = int(EstimatedATHour) -  24
        if(int(EstimatedATHour) == 0):
            EstimatedATHour = "00"
        elif(int(EstimatedATHour) < 10):
            EstimatedATHour = "0" + str(EstimatedATHour)
        else:
            EstimatedATHour = str(EstimatedATHour)
        if(int(EstimatedATMin) == 0):
            EstimatedATMin = "00"
        elif(int(EstimatedATMin) < 10):
            EstimatedATMin = "0" + str(EstimatedATMin)
        else:
            EstimatedATMin = str(EstimatedATMin)
        flight.update({'EstimatedATMin': EstimatedATMin})
        flight.update({'EstimatedATHour': EstimatedATHour})
        planned_arrtime = str(item['EstimatedATHour']) + str(item['EstimatedATMin'])
        flight.update({'planned_arrtime': planned_arrtime})

    #Calculate Planned Arrival time using SCH Dep time and Planned ERT
    def AirCalculation(flight):
        if int(flight["planned_tascruise"]) < 50:
            ##No TAS retarded Pilot
            flight["planned_tascruise"] = 350

        ERTCalc(flight)
        planned_arrtime = 0
        planned_arrhour = int(flight["ESTHOURERT"]) + int(str(now.strftime("%H%M"))[:-2])
        planned_arrmin = int(flight["ESTMINERT"]) + int(str(now.strftime("%H%M"))[-2:])
        if(planned_arrmin > 59):
            planned_arrmin -= 60
            planned_arrhour += 1
        if(int(planned_arrhour) > 23):
            planned_arrhour -= 24
        if(int(planned_arrhour) == 0):
            planned_arrhour = "00"
        elif(int(planned_arrhour) < 10):
            planned_arrhour = "0" + str(planned_arrhour)
        if(int(planned_arrmin) == 0):
            planned_arrmin = "00"
        if(int(planned_arrmin) < 10):
            planned_arrmin = "0" + str(planned_arrmin)
        planned_arrtime = str(planned_arrhour) + str(planned_arrmin)
        flight.update({'planned_arrtime': planned_arrtime})
        


    try:
        urllib.request.urlretrieve("http://cluster.data.vatsim.net/vatsim-data.json", "./vatsim-data.json")
    except:
        print("Unable to fetch VATSIM Data")
    with open("./vatsim-data.json", errors='ignore') as jsondata:
        data = json.load(jsondata)

    with open("./public/denver-data.json", errors='ignore') as activedendata:
        activedata = json.load(activedendata)



    online = data["clients"]
    for item in online:
        if item["clienttype"] == "ATC":
            online.remove(item)
    prefile = data["prefiles"]

    DENArr = []
    for item in online:
        if item["planned_destairport"] == "KDEN":
            DENArr.append(item)

    onlineCallsigns = []
    for item in DENArr:
        onlineCallsigns.append(item["callsign"])

    CallsignList = []
    for item in activedata:
        CallsignList.append(item["callsign"])
    for item in DENArr:
        if not item['callsign'] in CallsignList:
            item.update({'actual_deptime': 0})
            activedata.append(item)
        else:
            for flight in activedata:
                if(item['callsign'] == flight['callsign']):
                    flight.update({'latitude': item['latitude']})
                    flight.update({'longitude': item['longitude']})
                    flight.update({'groundspeed': item['groundspeed']})
    for item in activedata:
        if not item['callsign'] in onlineCallsigns:
            activedata.remove(item)


    def flightactive(flight):
        if(flight["distance"] < 5):
            flight.update({'status': "Arrived"})
        elif(flight['groundspeed'] > 50 and flight['actual_deptime'] == 0):
            flight.update({'actual_deptime': now.strftime("%H%M")})
        elif(flight['groundspeed'] > 50):
            flight.update({'status': "Active"})
        else:
            flight.update({'status': "Departing"})

    def Gate(flight):
        route = flight['planned_route'].upper()
        gate = ""
        if "AALLE" in route:
            gate = "AALLE"
        elif "LAWGR" in route:
            gate = "LAWGR"
        elif "CLASH" in route:
            gate = "CLASH"
        elif "NIIXX" in route:
            gate = "NIIXX"
        elif "TBARR" in route:
            gate = "TBARR"
        elif "SSKII" in route:
            gate = "SSKII"
        elif "LONGZ" in route:
            gate = "LONGZ"
        elif "FLATI" in route:
            gate = "FLATI"
        elif "KIPPR" in route:
            gate = "LANDR"
        elif "ANCHR" in route:
            gate = "LANDR"
        elif "LANDR" in route:
            gate = "LANDR"
        elif "JAGGR" in route:
            gate = "DANDD"
        elif "PURRL" in route:
            gate = "DANDD"
        elif "DANDD" in route:
            gate = "DANDD"
        elif "PEEKK" in route:
            gate = "LARKS"
        elif "LDORA" in route:
            gate = "LARKS"
        elif "LARKS" in route:
            gate = "LARKS"
        elif "TSHNR" in route:
            gate = "RAMMS"
        elif "MOLTN" in route:
            gate = "RAMMS"
        elif "RAMMS" in route:
            gate = "RAMMS"
        elif "KAILE" in route:
            gate = "TOMSN"
        elif "FRNCH" in route:
            gate = "TOMSN"
        elif "TOMSN" in route:
            gate = "TOMSN"
        elif "ZPLYN" in route:
            gate = "QUAIL"
        elif "BOSSS" in route:
            gate = "QUAIL"
        elif "QUAIL" in route:
            gate = "QUAIL"
        elif "TELLR" in route:
            gate = "POWDR"
        elif "CREDE" in route:
            gate = "POWDR"
        elif "POWDR" in route:
            gate = "POWDR"
        elif "KOHOE" in route:
            gate = "SAYGE"
        elif "WAHUU" in route:
            gate = "SAYGE"
        else:
            gate = "ERROR"
        flight.update({"gate": gate})

    def index(flight):
        if(int(flight['planned_arrtime']) < 100):
            hourarr = 0
        else:
            hourarr = int(flight['planned_arrtime'][:-2])
        hour = int(now.strftime("%H"))
        if(hour > hourarr):
            hour -= 24
        index = (hourarr - hour + 1)
        flight.update({'index': index})

    for item in activedata:
        if(item['groundspeed'] < 50):
            GroundCalculation(item)
        else:
            AirCalculation(item)
        flightactive(item)
        Gate(item)
        index(item)
    with open('./public/denver-data.json', 'w') as json_file:
	    json.dump(activedata, json_file)

    if(min == 0):
            with open('./public/AAR.txt', 'r') as json_file:
                    txt = json.loads(json_file.readline())
                    txt.pop(0)
                    txt.append(int(txt[len(txt)-1]))
                    f = open('./public/AAR.txt', 'w')
                    f.write(str(txt))
                    f.close()

    time.sleep(60)



public/index.php
<!DOCTYPE html>
<html>
        <head>
                <meta http-equiv="X-UA-Compatible" content="IE=11,10,9"/>
                <base href="."/>
                <title>AADC - KDEN</title>
                <meta name="viewport" content="width=device-width,initial-scale=1">
                <link href="styles.1a0f2c06eaf52ecc62cc.bundle.css" rel="stylesheet"/>
        </head>
        <body style="background-color:#FFFFFF">
                <aadc-app _nghost-c0="" ng-version="4.3.5">
                        <div _ngcontent-c0="" id="version"><a href="./changelog.txt">v2.1.0 - Raaj Patel</a></div>
                                <div _ngcontent-c0="" class="container-fluid">
                                        <div _ngcontent-c0="" class="row">
                                        <div _ngcontent-c0="" style="text-align: center;">
                                                <h3 _ngcontent-c0=""><img _ngcontent-c0="" height="80" src="assets/images/icon.png" width="80"><span _ngcontent-c0="" style="padding-left:20px;"></span>Airport Arrival Demand Chart</h3>
                                                        <text x="681" text-anchor="middle" class="highcharts-title" style="color:#333333;font-size:18px;fill:#333333;" y="24"><tspan>DEN</tspan></text>
                                                        <br>
                                                        <text x="681" text-anchor="middle" class="highcharts-subtitle" style="color:#666666;fill:#666666;" y="52"><tspan><span id="time"></span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Cancelled Flights: 0&nbsp;&nbsp;&nbsp;&nbsp;No GDP</tspan></text>
                                        </div>
                                        </div>
                        </div>
                </aadc-app>

                <div class="chart-container" style="position: relative; height:400px;">
                <canvas id="myChart"></canvas>
                </div>
                <div class="chart-container" style="position: relative; height:400px;">
                <canvas id="myChart2"></canvas>
                </div>
                <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
                <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/chart.js@2.9.3/dist/Chart.min.js"></script>
                <script type="text/javascript" src="denver-data.json"></script>
                <script type="text/javascript" src="main.js"></script>
        </body>
</html>



public/main.js
$(document).ready(function(){
var arr = [0, 0]

//const data = [{"callsign": "SWA354", "cid": "1459238", "realname": "Nick Haynie KGEG", "clienttype": "PILOT", "frequency": null, "latitude": 43.50939, "longitude": -116.01437, "altitude": 41078, "groundspeed": 493, "planned_aircraft": "B737", "planned_tascruise": "450", "planned_depairport": "KPDX", "planned_altitude": "41000", "planned_destairport": "KDEN", "server": "USA-WEST", "protrevision": 100, "rating": 1, "transponder": 2000, "facilitytype": 0, "visualrange": 0, "planned_revision": "0", "planned_flighttype": "I", "planned_deptime": "2055", "planned_actdeptime": "0", "planned_hrsenroute": "2", "planned_minenroute": "25", "planned_hrsfuel": "3", "planned_minfuel": "43", "planned_altairport": "KCOS", "planned_remarks": " /v/", "planned_route": "WHAMY4 IMB BOI OCS MJANE FLATI1", "planned_depairport_lat": 0.0, "planned_depairport_lon": 0.0, "planned_destairport_lat": 0.0, "planned_destairport_lon": 0.0, "atis_message": null, "time_last_atis_received": "2020-05-31T21:47:26.2704738Z", "time_logon": "2020-05-31T21:47:26.2704735Z", "heading": 120, "qnh_i_hg": 30.0, "qnh_mb": 1016, "actual_deptime": "2155", "distance": 553.07569496058, "ESTHOURERT": "1", "ESTMINERT": 14, "planned_arrtime": "2315", "gate": "FLATI", "index": 2, "status": "Active"}, {"callsign": "DAL6555", "cid": "1308631", "realname": "Pete Melomo KDEN", "clienttype": "PILOT", "frequency": null, "latitude": 39.85283, "longitude": -104.67766, "altitude": 7376, "groundspeed": 19, "planned_aircraft": "B738/L", "planned_tascruise": "463", "planned_depairport": "KMSP", "planned_altitude": "36000", "planned_destairport": "KDEN", "server": "USA-WEST", "protrevision": 100, "rating": 1, "transponder": 3660, "facilitytype": 0, "visualrange": 0, "planned_revision": "4", "planned_flighttype": "I", "planned_deptime": 2231, "planned_actdeptime": "0", "planned_hrsenroute": "1", "planned_minenroute": "36", "planned_hrsfuel": "3", "planned_minfuel": "14", "planned_altairport": "KEGE", "planned_remarks": " /v/", "planned_route": "SCHEP9 ONL PORDR AALLE1", "planned_depairport_lat": 0.0, "planned_depairport_lon": 0.0, "planned_destairport_lat": 0.0, "planned_destairport_lon": 0.0, "atis_message": null, "time_last_atis_received": "2020-05-31T21:47:28.2624728Z", "time_logon": "2020-05-31T21:47:28.2624724Z", "heading": 181, "qnh_i_hg": 30.06, "qnh_mb": 1018, "actual_deptime": "2155", "distance": 0.5695450923377646, "ESTHOURERT": "0", "ESTMINERT": 0, "planned_arrtime": "2231", "gate": "AALLE", "index": 1, "EstimatedATMin": "31", "EstimatedATHour": "22", "status": "Arrived"}, {"callsign": "CPA876", "cid": "1374884", "realname": "Daniel Lai VHHH", "clienttype": "PILOT", "frequency": null, "latitude": 35.50555, "longitude": 142.24493, "altitude": 33046, "groundspeed": 558, "planned_aircraft": "H/B77W/L", "planned_tascruise": "505", "planned_depairport": "VHHH", "planned_altitude": "31000", "planned_destairport": "KDEN", "server": "SINGAPORE", "protrevision": 100, "rating": 1, "transponder": 2200, "facilitytype": 0, "visualrange": 0, "planned_revision": "0", "planned_flighttype": "I", "planned_deptime": "1840", "planned_actdeptime": "0", "planned_hrsenroute": "13", "planned_minenroute": "15", "planned_hrsfuel": "14", "planned_minfuel": "55", "planned_altairport": "KCOS", "planned_remarks": "PBN/A1B1C1D1L1O1S2 DOF/200531 REG/BKPM EET/RCAA0034 RJJJ0123 KZAK0530 KZOA1100 KZLC1145 KZDV1236 OPR/CPA PER/D RALT/RJCM PACD KMFR RMK/TCAS SIMBRIEF /v/", "planned_route": "OCEAN2B OCEAN V3 ENVAR M750 LEKOS R583 BORDO Z29 GAKIA Y293 YULIA Y57 DOMBI/N0496F330 Y57 POROT OTR11 LEPKI 39E60 42E70 44E80/N0489F350 44N70 43N60 42N50 41N40/N0484F370 40N30 SHENU RBL BAM J154 TCH J56 CHE TOMSN8", "planned_depairport_lat": 0.0, "planned_depairport_lon": 0.0, "planned_destairport_lat": 0.0, "planned_destairport_lon": 0.0, "atis_message": null, "time_last_atis_received": "2020-05-31T21:47:30.5981371Z", "time_logon": "2020-05-31T21:47:30.5981367Z", "heading": 55, "qnh_i_hg": 29.97, "qnh_mb": 1015, "actual_deptime": "2155", "distance": 4964.643563640208, "ESTHOURERT": "9", "ESTMINERT": 50, "planned_arrtime": "0751", "gate": "TOMSN", "index": 10, "status": "Active"}, {"callsign": "AAL832", "cid": "933980", "realname": "John Mullen KCLT", "clienttype": "PILOT", "frequency": null, "latitude": 39.91173, "longitude": -104.68672, "altitude": 10254, "groundspeed": 150, "planned_aircraft": "H/B788/L", "planned_tascruise": "479", "planned_depairport": "KATL", "planned_altitude": "40000", "planned_destairport": "KDEN", "server": "USA-EAST", "protrevision": 100, "rating": 1, "transponder": 473, "facilitytype": 0, "visualrange": 0, "planned_revision": "3", "planned_flighttype": "I", "planned_deptime": "1920", "planned_actdeptime": "1920", "planned_hrsenroute": "2", "planned_minenroute": "35", "planned_hrsfuel": "3", "planned_minfuel": "57", "planned_altairport": "KEGE", "planned_remarks": "+VFPS+/V/WWW.JOINAVA.ORG", "planned_route": "CUTTN2 HANKO MEM RZC BVO GCK KIISS CLASH1", "planned_depairport_lat": 0.0, "planned_depairport_lon": 0.0, "planned_destairport_lat": 0.0, "planned_destairport_lon": 0.0, "atis_message": null, "time_last_atis_received": "2020-05-31T21:47:32.4812469Z", "time_logon": "2020-05-31T21:47:32.4812465Z", "heading": 188, "qnh_i_hg": 30.01, "qnh_mb": 1016, "actual_deptime": "2155", "distance": 3.069991603441191, "ESTHOURERT": "0", "ESTMINERT": 1, "planned_arrtime": "2202", "gate": "CLASH", "index": 1, "status": "Arrived"}, {"callsign": "DAL1470", "cid": "1329345", "realname": "zachary hazen KMSP", "clienttype": "PILOT", "frequency": null, "latitude": 41.00116, "longitude": -106.84836, "altitude": 37280, "groundspeed": 463, "planned_aircraft": "B738/L", "planned_tascruise": "452", "planned_depairport": "KSEA", "planned_altitude": "35000", "planned_destairport": "KDEN", "server": "USA-EAST", "protrevision": 100, "rating": 1, "transponder": 2200, "facilitytype": 0, "visualrange": 0, "planned_revision": "1", "planned_flighttype": "I", "planned_deptime": "1830", "planned_actdeptime": "1830", "planned_hrsenroute": "2", "planned_minenroute": "8", "planned_hrsfuel": "3", "planned_minfuel": "39", "planned_altairport": "KBFF", "planned_remarks": "+VFPS+/V/CHARTS ON BOARD", "planned_route": "SUMMA2 SUMMA J54 BKE KU81M PIH OCS MJANE FLATI1", "planned_depairport_lat": 0.0, "planned_depairport_lon": 0.0, "planned_destairport_lat": 0.0, "planned_destairport_lon": 0.0, "atis_message": null, "time_last_atis_received": "2020-05-31T21:47:33.8879386Z", "time_logon": "2020-05-31T21:47:33.8879384Z", "heading": 115, "qnh_i_hg": 30.22, "qnh_mb": 1023, "actual_deptime": "2155", "distance": 120.67127460378615, "ESTHOURERT": "0", "ESTMINERT": 16, "planned_arrtime": "2217", "gate": "FLATI", "index": 1, "status": "Active"}]

var fa = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var dep = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var RAMMS = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var TOMSN = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var LANDR = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var SAYGE = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var DANDD = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var QUAIL = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var LARKS = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var POWDR = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
//RNAV
var AALLE = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var LAWGR = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var CLASH = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var NIIXX = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var TBARR = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var SSKII = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var LONGZ = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var FLATI = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

var ERROR = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
var data = []
var AARNumbers

$.get('/denver-data.json', myData => {
	data = myData;
$.get('/AAR.txt', Data => {
	AARNumbers = JSON.parse(Data);

//data = JSON.parse(data)
//Array.prototype.forEach.call(data, flight => {
data.forEach(flight => {
    console.log(flight['callsign'] + " ETA:" + flight['planned_arrtime']+"z DEPICAO:"+ flight["planned_depairport"] + " ARRGate:" + flight["gate"])
    index = flight['index']
    gate = flight['gate']
    status = flight['status']
    if(status == "Departing"){
        dep[index] = dep[index] + 1
    }
    if(status == "Active"){
        fa[index] = fa[index] + 1
    }
    if(status == "Arrived"){
        arr[index] = arr[index] + 1
    }
    if(gate == "AALLE"){
        AALLE[index] = AALLE[index] +1
    }
    else if(gate == "LAWGR"){
        LAWGR[index] = LAWGR[index] +1
    }
    else if(gate == "CLASH"){
        CLASH[index] = CLASH[index] +1
    }
    else if(gate == "NIIXX"){
        NIIXX[index] = NIIXX[index] +1
    }
    else if(gate == "TBARR"){
        TBARR[index] = TBARR[index] +1
    }
    else if(gate == "SSKII"){
        SSKII[index] = SSKII[index] +1
    }
    else if(gate == "LONGZ"){
        LONGZ[index] = LONGZ[index] +1
    }
    else if(gate == "FLATI"){
        FLATI[index] = FLATI[index] +1
    }
    else if(gate == "ERROR"){
        ERROR[index] = ERROR[index] +1
    }
    else if(gate == "RAMMS"){
        RAMMS[index] = RAMMS[index] +1
    }
    else if(gate == "TOMSN"){
        TOMSN[index] = TOMSN[index] +1
    }
    else if(gate == "LANDR"){
        LANDR[index] = LANDR[index] +1
    }
    else if(gate == "SAYGE"){
        SAYGE[index] = SAYGE[index] +1
    }
    else if(gate == "DANDD"){
        DANDD[index] = DANDD[index] +1
    }
    else if(gate == "QUAIL"){
        QUAIL[index] = QUAIL[index] +1
    }
    else if(gate == "LARKS"){
        LARKS[index] = LARKS[index] +1
    }
    else if(gate == "POWDR"){
        POWDR[index] = POWDR[index] +1
    }
    else{
        ERROR[index] = ERROR[index] +1
    }
});

var ctx = document.getElementById('myChart');
var d = new Date();
var min = d.getUTCMinutes();
if(min < 10){
	min = '0'+min;
} else {
	min = min;
}
var hour = d.getUTCHours();
if(hour < 10){
        hour = '0'+hour;
} else {
        hour = hour;
}
document.getElementById("time").innerHTML = d.getUTCMonth()+1+"/"+d.getUTCDate()+"/"+d.getUTCFullYear()+" "+hour+":"+min+"z";
var UH = d.getUTCHours();
UH = UH-1;
if(UH < 0){
	UH = UH+24;
}
var lab = []
for(var i = 0; i<16; i++){
        var UT = UH+i;
        if(UT > 23){
                UT = UT-24;
        }
        if(UT.toString().length == 1){
                UT = "0" + UT
        }
        lab.push(UT+"00");
}
ctx.height = 400;
var myChart = new Chart(ctx, {
    type: 'bar',

    // The data for our dataset
    data: {
        labels: lab,
        datasets: [{
            label: 'Arrived',
            backgroundColor: '#000000',
            borderColor: '#000',
            data: arr,
                        barPercentage: 0.6

                }, {
            label: 'Flight Active',
            backgroundColor: '#ff0000',
            borderColor: '#000',
            data: fa,
                        barPercentage: 0.6

                }, {
            label: 'Departing',
            backgroundColor: '#00ff00',
            borderColor: '#000',
            data: dep,
                        barPercentage: 0.6

                }, {
                        label: 'Arrival Rate',
                        data: AARNumbers,
                        type: 'line',
                        backgroundColor: 'rgba(0,0,0, 0.01)',
                        borderColor: 'rgba(0, 0, 0, 1)',
                        pointRadius: 0,
                        pointHitRadius: 50,
			lineTension: 0
        }]
    },
    options: {
            maintainAspectRatio: false,
	    scales: {
                xAxes: [{
                stacked: true
            }],
            yAxes: [{
                ticks: {
                    suggestedMin: 0,
                    suggestedMax: 175,
                    stepSize: 25
                },
                                stacked: true
            }]
        }
    }
});

var ctx2 = document.getElementById('myChart2');
ctx2.height = 400;
var myChart = new Chart(ctx2, {
    type: 'bar',

    // The data for our dataset
    data: {
        datasets: [{
            label: 'RAMMS',
            backgroundColor: '#ff0000',
            borderColor: '#000',
            data: RAMMS,
                        barPercentage: 0.6

                }, {
            label: 'TOMSN',
            backgroundColor: '#ff8482',
            borderColor: '#000',
            data: TOMSN,
                        barPercentage: 0.6

                }, {
            label: 'LANDR',
            backgroundColor: '#ffbf00',
            borderColor: '#000',
            data: LANDR,
                        barPercentage: 0.6

                }, {
            label: 'SAYGE',
            backgroundColor: '#ffdc84',
            borderColor: '#000',
            data: SAYGE,
                        barPercentage: 0.6

                }, {
            label: 'DANDD',
            backgroundColor: '#00ff00',
            borderColor: '#000',
            data: DANDD,
                        barPercentage: 0.6

                }, {
            label: 'QUAIL',
            backgroundColor: '#85ff7a',
            borderColor: '#000',
            data: QUAIL,
                        barPercentage: 0.6

                }, {
            label: 'LARKS',
            backgroundColor: '#8902ff',
            borderColor: '#000',
            data: LARKS,
                        barPercentage: 0.6

                }, {
            label: 'POWDR',
            backgroundColor: '#bf75ff',
            borderColor: '#000',
            data: POWDR,
                        barPercentage: 0.6

                }, {
            label: 'AALLE',
            backgroundColor: '#ff6600',
            borderColor: '#000',
            data: AALLE,
                        barPercentage: 0.6

                }, {
            label: 'CLASH',
            backgroundColor: '#ffb273',
            borderColor: '#000',
            data: CLASH,
                        barPercentage: 0.6

                }, {
            label: 'LAWGR',
            backgroundColor: '#d0ff00',
            borderColor: '#000',
            data: LAWGR,
                        barPercentage: 0.6

                }, {
            label: 'NIIXX',
            backgroundColor: '#ebff6a',
            borderColor: '#000',
            data: NIIXX,
                        barPercentage: 0.6

                }, {
            label: 'TBARR',
            backgroundColor: '#045cff',
            borderColor: '#000',
            data: TBARR,
                        barPercentage: 0.6

                }, {
            label: 'SSKII',
            backgroundColor: '#83bdff',
            borderColor: '#000',
            data: SSKII,
                        barPercentage: 0.6

                }, {
            label: 'LONGZ',
            backgroundColor: '#ff00a2',
            borderColor: '#000',
            data: LONGZ,
                        barPercentage: 0.6

                }, {
            label: 'FLATI',
            backgroundColor: '#ff8bd8',
            borderColor: '#000',
            data: FLATI,
                        barPercentage: 0.6

                }, {
            label: 'OTHER',
            backgroundColor: '#000',
            borderColor: '#000',
            data: ERROR,
                        barPercentage: 0.6

                }],
                labels: lab
    },
    options: {
                maintainAspectRatio: false,
                scales: {
                        xAxes: [{
                stacked: true
            }],
            yAxes: [{
		ticks: {
		    suggestedMin: 0,
                    suggestedMax: 150
		},
                stacked: true
            }]
        }
    }
});

});

});

});
